lib/nat/service.go

```diff
@@ -314,7 +314,16 @@ func (s *Service) acquireNewLocked(ctx context.Context, mapping *Mapping, nats m
 func (s *Service) tryNATDevice(ctx context.Context, natd Device, intPort, extPort int, leaseTime time.Duration) (Address, error) {
 	var err error
 	var port int
-
+	// For IPv6, we just try to create the pinhole. If it fails, nothing can be done (probably no IGDv2 support).
+	// If it already exists, the relevant UPnP standard requires that the gateway recognizes this and updates the lease time.
+	// Since we usually have a global unicast IPv6 address so no conflicting mappings, we just request the port we're running on
+	if natd.IsIPv6GatewayDevice() {
+		_, err = natd.TryAddPinhole(ctx, TCP, intPort, "syncthing", leaseTime)
+		return Address{
+			natd.GetLocalIPv4Address(),
```

Comment:
Not sure if i'm reading this correctly but if we create a pinhole, that address is obtained via `natd.GetLocalIPv4Address()` ? That is a bit confusing.

Reply:
What are we doing with that single IPv6? Seems strange as we either are interested in all successfully pinholed addresses or none?

Reply:
Hm, fair enough, I thought there'd be more changes outside of the NAT service required if we tried handling all IPv6 addresses (which is definitely the better choice), but with the latest commit AddPinhole just returns a `[]net.IP` slice containing all successfully pinholed IPs.